src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -0,0 +1,359 @@
+                            log.warn("capacity exhausted at the moment %s", e);
+                        }
+                    }
+
+                    // observe if newly observed value in nodes increased from original snapshot
+                    Map<String, Integer> machineSetToReadyNodeCountSnapshotCurrent = FleetshardUtils.getReadyNodesPerEachMachineSetContainingName(oc, "kafka-standard");
+                    for (var currentlyObservedMachineSet  : machineSetToReadyNodeCountSnapshotCurrent.entrySet()) {
+                        // naming
+                        String machineSetName = currentlyObservedMachineSet.getKey();
+                        int newlyObservedValue = currentlyObservedMachineSet.getValue();
+                        int oldObservedValue = machineSetToReadyNodeCountSnapshotBefore.get(currentlyObservedMachineSet.getKey());
+
+                        // observing if value changed
+                        log.debug("machineSet '{}' capacity before '{}' and now '{}'", machineSetName, oldObservedValue, newlyObservedValue);
+                        if (newlyObservedValue > oldObservedValue) {
+                            log.info("MachineSet '{}' changed scaled number of its nodes", machineSetName);
+                            return true;
+                        }
+                    }
+
+                    // observe if any of newly crated kafka instance really is ready state (node for sure scaled), otherwise continue waiting
+                    return kafkaMgmtApi
+                        .getKafkas(null, null, null, null)
+                        .getItems().stream()
+                        .filter(e -> e.getName().contains(INSTANCE_NAME_PREFIX))
+                        .filter(e -> e.getOwner().equals(Environment.PRIMARY_USERNAME))
+                        .anyMatch(e -> e.getStatus().equals("ready"));
+                }
+            );
+
+        } catch (Exception ignored) {
+        } finally {
+            // delete and wait for cleaning of all instances spawned
+            KafkaMgmtApiUtils.deleteSearchedKafkaInstancesByOwner(kafkaMgmtApi, INSTANCE_NAME_PREFIX, Environment.PRIMARY_USERNAME);
+        }
+    }
+
+    @Test(dataProvider = "managedKafkaTypes")
+    @SneakyThrows
+    public void testReportedCapacityMatchesNumberOfExistingInstances(String mkType) {
+
+        int observedRemainingCapacityInitially = FleetshardUtils.getCapacityRemainingUnitsFromMKAgent(oc, ManagedKafkaType.valueOf(mkType));
+        log.info("observed remaining capacity before creating new instance '{}'", observedRemainingCapacityInitially);
+        if (observedRemainingCapacityInitially == 0)
+            throw new SkipException("cluster has no free capacity to test at the moment");
+
+        Set<String> kafkaInstanceNamesBeforeCreating = FleetshardUtils.listManagedKafka(oc, ManagedKafkaType.valueOf(mkType)).stream()
+            .map(e -> e.getMetadata().getName())
+            .collect(Collectors.toSet());
```

Comment:
The list here will include reserved deployments as well.

Reply:
yeah but it always exist so it does not change the logic.

Reply:
+1